File: ebean-querybean/src/test/java/org/example/domain/MyInnerEmb.java
package org.example.domain;

import jakarta.persistence.Embeddable;
import jakarta.persistence.Embedded;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;

@Entity
public class MyInnerEmb {

  @Embeddable
  public static class EmbAddr {
    final long streetNum;
    final String line1;

    public EmbAddr(long streetNum, String line1) {
      this.streetNum = streetNum;
      this.line1 = line1;
    }
  }

  @Id
  long id;
  String one;

  @Embedded
  EmbAddr address;

  public long id() {
    return id;
  }

  public MyInnerEmb id(long id) {
    this.id = id;
    return this;
  }

  public String one() {
    return one;
  }

  public MyInnerEmb one(String beanOne) {
    this.one = beanOne;
    return this;
  }

  public EmbAddr address() {
    return address;
  }

  public MyInnerEmb setAddress(EmbAddr address) {
    this.address = address;
    return this;
  }
}

File: tests/test-kotlin/src/main/kotlin/org/example/domain/MyInnerEmb.kt
package org.example.domain

import jakarta.persistence.*

@Entity
@Table(name = "t_inner2")
class MyKInnerEmb {

  @Id
  var one: Long = 0
  var two: String = "0"
  var address: EmbAddre? = null

  @Embeddable
  data class EmbAddre(val line1: String, val line2: String)

}
